pcsx2/gui/MainMenuClicks.cpp

```diff
@@ -902,31 +902,32 @@ void MainEmuFrame::Menu_Recording_Play_Click(wxCommandEvent &event)
 	const bool initiallyPaused = g_InputRecordingControls.IsRecordingPaused();
 	if (!initiallyPaused)
 		g_InputRecordingControls.PauseImmediately();
+
 	wxFileDialog openFileDialog(this, _("Select P2M2 record file."), L"", L"",
 		L"p2m2 file(*.p2m2)|*.p2m2", wxFD_OPEN);
-	if (openFileDialog.ShowModal() == wxID_CANCEL)
+	if (openFileDialog.ShowModal() == wxID_CANCEL && !initiallyPaused)
 	{
-		if (!initiallyPaused)
-			g_InputRecordingControls.Resume();
+		g_InputRecordingControls.Resume();
 		return;
 	}
 
 	wxString path = openFileDialog.GetPath();
-	const bool recordingLoaded = g_InputRecording.IsActive();
-	if (!g_InputRecording.Play(path))
+	const bool wasRecordingActive = g_InputRecording.IsActive();
+	// TODO (future PR) - a warning about closing active recording would be a good idea
+	const bool playSuccessful = g_InputRecording.Play(path);
+
+	if (!playSuccessful && !initiallyPaused)
```

Comment:
Similar issue here. Even if loading failed, as long as the user manually paused before entering the file dialog, everything below will still proceed as if the loading had succeeded. Specifically, the part where it will sometimes resume playback.
(I called Stop_click() also for a future multitap-related reason).

Reply:
Hmm, can you elaborate on the future multi-tap related reason?  I see no point in the GUI having to call `Stop` if `Play` has failed.  The InputRecording code should properly handle such behaviour, it shouldn't be a burden on the GUI code.

Reply:
As it stands now, a controller on a multitap can only be read in if two things are true:
1. The pad plugin has that multitap active
2. Core also has that multitap active.
While I can't currently do anything about the PAD plugin setting, I can mess with the core setting (at least temporarily for the recording). So on load, if a multitap is required, it will 1. save the boolean setting of said multitap from before any recordings, and 2. set the multitap accordingly.
Additionally, the only virtual pads that should be available are those that should/can be polled from at any given time. In the case of a recording, the only virtual pads that will be modifiable are those that the recording specifically uses. Every other pad will have its menu option disabled for the duration of the recording.
If afterwards a user wants to stop recording, or is put in a situation where a recording _must_ be closed, pcsx2 must reset the new/stop/play menu options, all pad options, all multitap settings, and close the recording. Putting all of that in one function just makes sense in that regard.
Old rendition of this:
```
void MainEmuFrame::Menu_Recording_Stop_Click(wxCommandEvent &event)
{
	g_InputRecording.Stop();
	//Reenables all disabled menu options, setting multitaps to the buffer values
	m_menuRecording.FindChildItem(MenuId_Recording_New)->Enable(true);
	m_menuRecording.FindChildItem(MenuId_Recording_Stop)->Enable(false);
	m_menuConfig.FindChildItem(MenuId_Config_Multitap0Toggle)->Enable(true);
	m_menuConfig.FindChildItem(MenuId_Config_Multitap1Toggle)->Enable(true);

	//When no file is loaded, controller 1 of both ports will always be active
	m_menuRecording.FindChildItem(MenuId_Recording_VirtualPad_Port0)->Enable(true);
	m_submenuvirtualPort0.FindChildItem(MenuId_Recording_VirtualPad_Port0_1)->Enable(true);
	if (g_InputRecording.GetMultitapBuffer(0))
	{
		g_Conf->EmuOptions.MultitapPort0_Enabled = true;
		m_submenuvirtualPort0.FindChildItem(MenuId_Recording_VirtualPad_Port0_2)->Enable(true);
		m_submenuvirtualPort0.FindChildItem(MenuId_Recording_VirtualPad_Port0_3)->Enable(true);
		m_submenuvirtualPort0.FindChildItem(MenuId_Recording_VirtualPad_Port0_4)->Enable(true);
	}
	else if (g_Conf->EmuOptions.MultitapPort0_Enabled)
	{
		g_Conf->EmuOptions.MultitapPort0_Enabled = false;
		m_submenuvirtualPort0.FindChildItem(MenuId_Recording_VirtualPad_Port0_2)->Enable(false);
		m_submenuvirtualPort0.FindChildItem(MenuId_Recording_VirtualPad_Port0_3)->Enable(false);
		m_submenuvirtualPort0.FindChildItem(MenuId_Recording_VirtualPad_Port0_4)->Enable(false);
	}

	m_menuRecording.FindChildItem(MenuId_Recording_VirtualPad_Port1)->Enable(true);
	m_submenuvirtualPort1.FindChildItem(MenuId_Recording_VirtualPad_Port1_1)->Enable(true);
	if (g_InputRecording.GetMultitapBuffer(1))
	{
		g_Conf->EmuOptions.MultitapPort1_Enabled = true;
		m_submenuvirtualPort1.FindChildItem(MenuId_Recording_VirtualPad_Port1_2)->Enable(true);
		m_submenuvirtualPort1.FindChildItem(MenuId_Recording_VirtualPad_Port1_3)->Enable(true);
		m_submenuvirtualPort1.FindChildItem(MenuId_Recording_VirtualPad_Port1_4)->Enable(true);
	}
	else if (g_Conf->EmuOptions.MultitapPort1_Enabled)
	{
		g_Conf->EmuOptions.MultitapPort1_Enabled = false;
		m_submenuvirtualPort1.FindChildItem(MenuId_Recording_VirtualPad_Port1_2)->Enable(false);
		m_submenuvirtualPort1.FindChildItem(MenuId_Recording_VirtualPad_Port1_3)->Enable(false);
		m_submenuvirtualPort1.FindChildItem(MenuId_Recording_VirtualPad_Port1_4)->Enable(false);
	}
	// Displays original multitap settings in config menu
	ApplySettings();
}
```

Reply:
I see.  Off the top of my head I think it would be a lot more simplified to pull the handling of the menu items out into a generic function, rather than hard-coding `true/false` so it can be called from `Stop()` / `Play()` and update the items accordingly.  That way `Play` isn't calling stop even though the recording has already been properly `Stop`'d

But without actually taking a stab at that refactor myself, this isn't really strong advice.  Is there any reason for me to add back this redundant call as the multi-tap implementation isn't being added in this PR/Branch?

Reply:
A different implementation, one I actually was gonna do at some point, was to pull the menu changing stuff into a different no-parameter MainEmuFrame function altogether. Then we could leave Stop_Click() to essentially just:
```
void MainEmuFrame::Menu_Recording_Stop_Click(wxCommandEvent &event)
{
	g_InputRecording.Stop();
	newFunction();
}
```